lib/medtronic/medtronicSimulator.js

```diff
@@ -239,42 +239,46 @@ exports.make = function(config){
             event.suppressed.scheduleName = currPumpSettings.activeSchedule;
           }
           if(currBasal.deliveryType === 'temp') {
-            // nest suppressed scheduled basal in temp basal
+            // nest suppressed scheduled basal in temp basal inside suspended basal
             event.suppressed.suppressed = currBasal.suppressed;
 
-            // also, restart temp basal on resume
-            if(currStatus && currStatus.index === event.index) {
-              // if the indexes are the same, the suspended basal was
-              // created from the same record as the suspend/resume event
-              event.duration = currStatus.duration;
-
-              var resumeBasal = _.clone(currBasal);
-              resumeBasal.time = new Date(Date.parse(currStatus.time) + currStatus.duration).toISOString();
-              resumeBasal.deviceTime = sundial.formatDeviceTime(Date.parse(currStatus.deviceTime) + currStatus.duration);
-              resumeBasal.index = event.resumeIndex;
-              delete resumeBasal.duration; // we don't know the new duration yet
-
-              // finish up the temp basal before the suspend
-              if(currBasal.duration) {
-                currBasal.with_expectedDuration(currBasal.duration);
-                var duration = Date.parse(event.time) - Date.parse(currBasal.time);
-                currBasal.with_duration(duration);
+            if(currBasal.duration > (Date.parse(currStatus.time) - Date.parse(currBasal.time) + currStatus.duration)) {
```

Comment:
Trying to check my understanding here...

`Date.parse(currStatus.time) - Date.parse(currBasal.time)` is the duration of the `temp` basal prior to the suspension...

So the total of `(Date.parse(currStatus.time) - Date.parse(currBasal.time) + currStatus.duration)` is the duration of the temp prior to suspend and the duration of the suspend, yes?

If you can think of a good variable name, that whole chunk could be more readable pulled out prior to the `if`, like:

```JavaScript
var tempBeforeSuspensionAndSuspensionDuration = (Date.parse(currStatus.time) - Date.parse(currBasal.time) + currStatus.duration);
if (currBasal.duration > tempBeforeSuspensionAndSuspensionDuration)
```

But y'know, only if you can come up with a better variable name than the one I just did ;)